datadotworld/datadotworld.py

```diff
@@ -193,9 +201,8 @@ def load_dataset(self, dataset_key, force_update=False, auto_update=False):
                             'function with the argument '
                             'auto_update=True or '
                             'force_update=True'.format(dataset_key))
-            except RestApiError:
-                # Not a critical step
-                pass
+            except RestApiError as e:
+                raise
```

Comment:
Let's think about what's going on here, for a minute when a `RestApiError` occurs. It can only mean that https://github.com/datadotworld/data.world-py/pull/80/files#diff-31fc69a07d5fac6aa1efbc356d841e14R179 failed. If/when that happens, the user still has a local version of the dataset that they can use, therefore, it makes sense for us to ignore (or warn), but continue to move along.

The best thing to do here would be to move this `except` block right after https://github.com/datadotworld/data.world-py/pull/80/files#diff-31fc69a07d5fac6aa1efbc356d841e14R179, to make this code easier to understand. And, if a `RestApiError` occurs, simply return `return LocalDataset(descriptor_file)` from its `except` block.

What do you think?

Reply:
I think that is right and makes the code clearer. Since the `except` is basically for `get_dataset` api call if it fails, It makes sense to wrap the try/except block closer to the relevant statement block.